Source/CarthageKit/Project.swift

```diff
@@ -584,10 +584,13 @@ public final class Project {
 							}
 							return self.downloadMatchingBinaries(for: dependency, atRevision: revision, fromRepository: repository, client: Client(server: server, isAuthenticated: false))
 						}
-						.flatMap(.concat) { self.unarchiveAndCopyBinaryFrameworks(zipFile: $0, projectName: dependency.name, commitish: revision, toolchain: toolchain) }
-						.on(completed: {
-							_ = try? FileManager.default.trashItem(at: checkoutDirectoryURL, resultingItemURL: nil)
-						})
+						.flatMap(.concat) {
+							return self.unarchiveAndCopyBinaryFrameworks(zipFile: $0, projectName: dependency.name, commitish: revision, toolchain: toolchain)
+								.on(terminated: {
+									// We can receive the 'interrupted' event due to the 'take(first: 1)' below, so listen to 'terminated' to ensure we catch it
+									_ = try? FileManager.default.removeItem(at: checkoutDirectoryURL)
```

Comment:
I think this used `trashItem` because we're trashing the checkouts folder. If you make local modifications and then run `carthage update`, your changes will be gone. Trashing them lets people recover them. I think it's good to be a little forgiving in this scenario.